dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -56,64 +69,104 @@ public void CollectionNameCantBeNullOrEmpty()
     [MemberData(nameof(LifetimesAndServiceKeys))]
     public virtual void CanRegisterVectorStore(ServiceLifetime lifetime, object? serviceKey)
     {
-        IServiceCollection services = this.CreateServices(serviceKey);
+        foreach (var registrationDelegate in this.StoreDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
 
-        this.RegisterVectorStore(services, lifetime, serviceKey);
+            registrationDelegate(services, serviceKey, lifetime);
 
-        using ServiceProvider serviceProvider = services.BuildServiceProvider();
-        // let's ensure that concrete types are registered
-        Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
-        // and the abstraction too
-        Verify<VectorStore>(serviceProvider, lifetime, serviceKey);
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // let's ensure that concrete types are registered
+            Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
+            // and the abstraction too
+            Verify<VectorStore>(serviceProvider, lifetime, serviceKey);
+        }
     }
 
     [Theory]
     [MemberData(nameof(LifetimesAndServiceKeys))]
     public void CanRegisterCollections(ServiceLifetime lifetime, object? serviceKey)
     {
-        IServiceCollection services = this.CreateServices(serviceKey);
+        foreach (var registrationDelegate in this.CollectionDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
 
-        this.RegisterCollection(services, lifetime, serviceKey: serviceKey);
+            registrationDelegate(services, serviceKey, "collectionName", lifetime);
 
-        using ServiceProvider serviceProvider = services.BuildServiceProvider();
-        // Let's ensure that concrete types are registered.
-        Verify<TCollection>(serviceProvider, lifetime, serviceKey);
-        // And the VectorStoreCollection abstraction.
-        Verify<VectorStoreCollection<TKey, TRecord>>(serviceProvider, lifetime, serviceKey);
-        // And the IVectorSearchable abstraction.
-        Verify<IVectorSearchable<TRecord>>(serviceProvider, lifetime, serviceKey);
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // Let's ensure that concrete types are registered.
```

Comment:
FWIW the code seems pretty self-explanatory to me, not sure the comments are needed.